hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataAvroResolver.java

```diff
@@ -221,11 +220,38 @@ private static void validate(Schema schema, Map<QueryPath, MappingField> fieldsB
                 throw new IllegalArgumentException("Field '" + path + "' does not exist in schema");
             }
 
-            Schema.Type schemaFieldType = unwrapNullableType(schemaField.schema()).getType();
-            if (!conversions.contains(schemaFieldType)) {
+            Schema fieldSchema = unwrapNullableType(schemaField.schema());
+            Schema.Type schemaFieldType = fieldSchema.getType();
+            if (!(mappingFieldTypeFamily == QueryDataTypeFamily.OBJECT
```

Comment:
nested ternary operator in `if` with negation?! It takes long to even parse that mentally. Could you create variables for the stages of this condition (which might also address Ivan's comment about meaning of those)?

Reply:
407f4405